core/src/datayoga_core/blocks/relational/utils.py

```diff
@@ -77,9 +78,34 @@ def get_engine(connection_name: str, context: Context, autocommit: bool = True)
         # add NLS_DATE_FORMAT for Oracle to explicitly set the date format to ISO
         sa.event.listen(engine, "connect", alter_session_on_oracle_connect)
 
+    if engine.driver == "pymysql":
+        # add a listener to configure PyMySQL dialect settings
+        sa.event.listen(engine, "connect", create_pymysql_dialect_configuration(engine))
     return engine, db_type
 
 
+def create_pymysql_dialect_configuration(engine: Engine) -> Callable[[DBAPIConnection, ConnectionPoolEntry], None]:
+    """Creates a function to configure the PyMySQL dialect.
+
+    Args:
+        engine (Engine): The SQLAlchemy engine to configure.
+
+    Returns:
+        Callable[[DBAPIConnection, ConnectionPoolEntry], None]: A function that configures PyMySQL dialect settings.
+    """
+    def configure_pymysql_dialect(dbapi_connection: DBAPIConnection, connection_record: ConnectionPoolEntry):
```

Comment:
Why do you need a submethod in there?

Reply:
just because it's tied with the `create_pymysql_dialect_configuration` that is needed to pass the engine within.